File: impl/mono/src/Formatters/Docker/FormatterDocker.cs
using System;
using System.Text;
using cb.Slices;

namespace cb.Formatters.Docker
{
    public class FormatterDocker : IFormatter
    {
        public void Write(Slice slice, StringBuilder sb)
        {
            foreach (var section in slice.Sections)
            {
                Write(section, sb);
            }
        }

        public void Write(SliceSection section, StringBuilder sb)
        {
            if (section.SectionType == SliceSection.Type.FROM)
            {
                foreach (var line in section.Lines)
                {
                    sb.Append("FROM ").AppendLine(line);
                }
            }

            if (section.SectionType == SliceSection.Type.RUN)
            {
                var text = string.Join($" && {"\\"} {Environment.NewLine}", section.Lines);
                sb.Append("RUN ");
                sb.AppendLine(text);
            }
        }
    }
}

File: impl/rust/src/formatters/formatter_docker.rs
//FormatterDocker Mod
use super::slice::{item, section};
// When this function is called, it will have to be changed to handle a return value, instead.
pub fn Write(slice: Slice) -> String { 
    let mut string = "";
    for i in slice.Section {
        if i.0 != None {
           string = format!("{}{}", string, section_to_string(i.0));
        }
    }
        string;
}
// Equivalent of the second "write" function in the FormatterDocker.cs file in the mono impl. In
// rust, however, functions cannot be overloaded.  
fn section_to_string(slice_section:Section, string:String) -> String {
   if slice_section.kind == (Some(Kind::From)) {
        for i in slice_section.items {
           string =  format!("FROM {}", i);
        }
   }
   if slice_section.kind == (Some(Kind::RUN)) {
        let mut text = "";
            for i in slice_section.items {
                text = format!("{}  &&  \\\\ \n{}", text, i);
            }
            // Above code is supposed to emulate the below line. I am unsure about it, though.
            // var text = string.Join($" && {"\\"} {Environment.NewLine}", section.Lines);

        string = format!("{}RUN \n{}\n", string, text);
 
   }
    return string;
}
